ONNX: Renaming scales parameter for interpolate

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31670 ONNX: Renaming scales parameter for interpolate**

This does the following:
1) Makes the float parameters to upsample optional (float?)
2) Renames and swaps the meaning of use_scale_factor to recompute_scale_factor as a more accurate name
3) Clarifies some documentation

Differential Revision: [D19240239](https://our.internmc.facebook.com/intern/diff/D19240239)

gchanan committed:
ONNX: Renaming scales parameter for interpolate

This does the following:
1) Makes the float parameters to upsample optional (float?)
2) Renames and swaps the meaning of use_scale_factor to recompute_scale_factor as a more accurate name
3) Clarifies some documentation

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit ad2eccf8:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=ad2eccf810ba6df2c9e60a2e5941c5301041b0ec).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4078000) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78731904) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78731904))
<details>
<summary>
<code>Dec 27 23:21:25 Failed to run '['/var/lib/jenkins/workspace/xla/scripts/generate_code.sh']'</code>
</summary>

```
Dec 27 23:21:25 AtenXlaType function missed override: Tensor upsample_bilinear2d(const Tensor& self, IntArrayRef output_size, bool align_corners, double scales_1, double scales_2); // upsample_bilinear2d(Tensor,IntArrayRef,bool,double,double)->Tensor 
Dec 27 23:21:25 AtenXlaType function missed override: Tensor upsample_bilinear2d_backward(const Tensor& grad_output, IntArrayRef output_size, IntArrayRef input_size, bool align_corners, double scales_1, double scales_2); // upsample_bilinear2d_backward(Tensor,IntArrayRef,IntArrayRef,bool,double,double)->Tensor 
Dec 27 23:21:25 AtenXlaType function missed override: Tensor upsample_nearest2d(const Tensor& self, IntArrayRef output_size, double scales_1, double scales_2); // upsample_nearest2d(Tensor,IntArrayRef,double,double)->Tensor 
Dec 27 23:21:25 AtenXlaType function missed override: Tensor upsample_nearest2d_backward(const Tensor& grad_output, IntArrayRef output_size, IntArrayRef input_size, double scales_1, double scales_2); // upsample_nearest2d_backward(Tensor,IntArrayRef,IntArrayRef,double,double)->Tensor 
Dec 27 23:21:25 Traceback (most recent call last): 
Dec 27 23:21:25   File "/var/lib/jenkins/workspace/xla/scripts/gen.py", line 1052, in <module> 
Dec 27 23:21:25     generate(args) 
Dec 27 23:21:25   File "/var/lib/jenkins/workspace/xla/scripts/gen.py", line 1022, in generate 
Dec 27 23:21:25     assert check_overrides(overrides, overridden) 
Dec 27 23:21:25 AssertionError 
Dec 27 23:21:25 Failed to run '['/var/lib/jenkins/workspace/xla/scripts/generate_code.sh']' 
Dec 27 23:21:25 Building torch_xla version: 0.8 
Dec 27 23:21:25 + cleanup 
Dec 27 23:21:25 + retcode=1 
Dec 27 23:21:25 + set +x 
Dec 27 23:21:25 =================== sccache compilation log =================== 
Dec 27 23:21:25 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 27 23:21:25 Compile requests              2524 
Dec 27 23:21:25 Compile requests executed     2267 
Dec 27 23:21:25 Cache hits                    2253 
Dec 27 23:21:25 Cache misses                     1 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

gchanan committed:
Update on "ONNX: Renaming scales parameter for interpolate"


This does the following:
1) Makes the float parameters to upsample optional (float?)
2) Renames and swaps the meaning of use_scale_factor to recompute_scale_factor as a more accurate name
3) Clarifies some documentation

Differential Revision: [D19240239](https://our.internmc.facebook.com/intern/diff/D19240239)

[ghstack-poisoned]
